src/Api.Rest.Dtos.Tests/Data/AttributeDtoTest.cs

```diff
@@ -1,295 +1,1068 @@
+		yield return CreateTestCase( localTime,      null, null, "2015-03-09T20:12:00+01:00", universalTime, localTime,      $"local: {universalTime.ToString( CultureInfo.InvariantCulture )}" );
+		yield return CreateTestCase( unspecifiedTime,null, null, "2015-03-09T19:12:00",       universalTime, unspecifiedTime, $"unspecified: {universalTime.ToString( CultureInfo.InvariantCulture )}" );
+
+		// @formatter:on — enable formatter after this line
+	}
+
+	private static IEnumerable CreateCatalogEntryAttributeTestCases()
+	{
+		// @formatter:off — disable formatter after this line
+
+		// value: date
+		var catalogEntry = new CatalogEntryDto{ Key = 4 };
+
+		yield return CreateTestCase( catalogEntry, 4, 4,       "4", null,catalogEntry, "catalog entry" );
+
+		// @formatter:on — enable formatter after this line
+	}
+
+	private static AttributeConversionTestCase CreateTestCase(
+		object attributeValue,
+		int? expectedIntValue, double? expectedDoubleValue, string expectedStringValue, DateTime? expectedDateValue,
+		object expectedRawValue,
+		string displayText = null )
+	{
+		return new AttributeConversionTestCase
 		{
-			var attribute1 = new AttributeDto( 1, "foo" );
-			var attribute2 = new AttributeDto( 1, "foo" );
+			Attribute = new AttributeDto( 1, attributeValue ),
+			ExpectedIntValue = expectedIntValue,
+			ExpectedDoubleValue = expectedDoubleValue,
+			ExpectedStringValue = expectedStringValue,
+			ExpectedDateValue = expectedDateValue,
+			ExpectedRawValue = expectedRawValue,
+			DisplayText = displayText
+		};
+	}
 
-			var hash1 = attribute1.GetHashCode();
-			var hash2 = attribute2.GetHashCode();
+	#endregion
 
-			hash1.Should().Be( hash2 );
-		}
+	#region class AttributeComparisonTestCase
+
+	public class AttributeComparisonTestCase
+	{
+		#region properties
+
+		public AttributeDto X { get; set; }
 
-		[Test]
-		public void GetHashCode_DifferentValue_DifferentHashes()
+		public AttributeDto Y { get; set; }
+
+		public bool ExpectedResult { get; set; }
+
+		#endregion
+
+		#region methods
+
+		public override string ToString()
 		{
-			var attribute1 = new AttributeDto( 1, "foo" );
-			var attribute2 = new AttributeDto( 1, "bar" );
+			return $"{PrintAttribute( X )}, {PrintAttribute( Y )}";
 
-			var hash1 = attribute1.GetHashCode();
-			var hash2 = attribute2.GetHashCode();
+			string PrintAttribute( AttributeDto attribute )
+			{
+				if( attribute.RawValue is CatalogEntryDto catalogEntry )
+					return $"Catalog entry {catalogEntry.Key}: {catalogEntry}";
 
-			hash1.Should().NotBe( hash2 );
+				return attribute.Value ?? "null";
+			}
 		}
 
-		[Test]
-		public void GetString_ValidAttribute_CorrectString()
-		{
-			var attribute = new AttributeDto( 1, "foo" );
+		#endregion
+	}
+
+	#endregion
+
+	#region class AttributeConversionTestCase
+
+	public class AttributeConversionTestCase
```

Comment:
This class is used in both AttributeDtoTest and AttributeDtoExtensionsTest, so it think you could extract it to its own file.

Reply:
Since the tested scenarios are basically the same, I followed your suggestion and extracted the class into its own file.